### ISSUE TITLE
'Gitlab' is not exported from '@gitbeaker/node'.

### ISSUE THREAD

**Issue by inkidu** [jdalrymple/gitbeaker#698]
**Description**

I am working on a react dashboard trying to import the git client from @gitbeaker/node as follows:

`import { Gitlab } from '@gitbeaker/node';`

I get the following warning in the editor:

```
module "d:/_dev/pcte/vdk-dashboard/node_modules/@gitbeaker/node/dist/index"
Could not find a declaration file for module '@gitbeaker/node'. 'd:/_dev/pcte/vdk-dashboard/node_modules/@gitbeaker/node/dist/index.js' implicitly has an 'any' type.
  Try `npm install @types/gitbeaker__node` if it exists or add a new declaration (.d.ts) file containing `declare module '@gitbeaker/node';`ts(7016)
```
I saw this with v18.0.0 and the previous version.
My version of node is 13.12.0.
I tried the workaround for this issue #304, but that didn't work for me.

**Steps to reproduce**

When I run the service (npm start) I get the following output:
```
./src/services/gitlabClient.js
Attempted import error: 'Gitlab' is not exported from '@gitbeaker/node'.
```

Here is my package.json:
```
{
  "name": "react-admin-dashboard",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "@gitbeaker/node": "^18.0.0",
    "@testing-library/jest-dom": "^4.2.4",
    "@testing-library/react": "^9.5.0",
    "@testing-library/user-event": "^7.2.1",
    "aphrodite": "^2.4.0",
    "react": "^16.13.1",
    "react-dom": "^16.13.1",
    "react-scripts": "3.4.1",
    "react-svg-line-chart": "^2.0.2",
    "simple-flexbox": "^2.3.0"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test",
    "eject": "react-scripts eject"
  },
  "eslintConfig": {
    "extends": "react-app"
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  }
}
```

**Expected behaviour**
I expect there to be a Gitlab class exported from @gitbeaker/node. I have also tried Projects and ProjectsBundle with no luck.

**Actual behaviour**
./src/services/gitlabClient.js
Attempted import error: 'Gitlab' is not exported from '@gitbeaker/node'.

**Possible fixes**
I am pretty sure it is me and not you, but maybe this could help your documentation.

**Comment by jdalrymple** [comment]
Seems odd but sound very similar to #686. Ill give it a quick look tomorrow. It could be an error in the export?

**Comment by inkidu** [comment]
Thank you so much.

**Comment by inkidu** [comment]
I rolled back to the old gitlab library 
`npm install gitlab` 
and it worked with that.

**Comment by jdalrymple** [comment]
Found the problem. Due to the way it was setup, only the default export is available. Im fixing that up now and it should be released in the next day or so

**Comment by inkidu** [comment]
I tried the workaround and got the following results.
'''
import Gitbeaker from '@gitbeaker/node';
//const { Gitlab } = Gitbeaker;

export default class GitlabClient {

    static instance = null;
    static getInstance() {
        if (this.instance === null) {
            this.instance = new GitlabClient();
        }
        return this.instance;
    }

    constructor() {
        this.config = require('../config.json');
        // this.api = new Gitlab ({
        //     host: this.config.gitlabURL,
        //     token: this.config.gitlabAuthToken,
        // });
        console.log('GitlabClient created!');

        return this;
    }
    
    // async getProjects() {
    //     let projects = await this.api.Projects.all({maxPages: 2});
    //     return projects;
    // }
}
'''
As soon as I uncomment the import Gitbeaker from '@gitbeaker/node'; line I get the following error stack:
'''
util.js:601 Uncaught TypeError: The "original" argument must be of type Function
    at Object.promisify (util.js:601)
    at Object.<anonymous> (get-body-size.js:7)
    at Object../node_modules/got/dist/source/utils/get-body-size.js (get-body-size.js:30)
    at __webpack_require__ (bootstrap:784)
    at fn (bootstrap:150)
    at Object.<anonymous> (normalize-arguments.js:15)
    at Object../node_modules/got/dist/source/normalize-arguments.js (normalize-arguments.js:441)
    at __webpack_require__ (bootstrap:784)
    at fn (bootstrap:150)
    at Object.<anonymous> (as-promise.js:8)
    at Object../node_modules/got/dist/source/as-promise.js (as-promise.js:152)
    at __webpack_require__ (bootstrap:784)
    at fn (bootstrap:150)
    at Object../node_modules/got/dist/source/create.js (create.js:4)
    at __webpack_require__ (bootstrap:784)
    at fn (bootstrap:150)
    at Object../node_modules/got/dist/source/index.js (index.js:7)
    at __webpack_require__ (bootstrap:784)
    at fn (bootstrap:150)
    at Module.<anonymous> (index.es.js:1)
    at Module../node_modules/@gitbeaker/node/dist/index.es.js (index.es.js:15)
    at __webpack_require__ (bootstrap:784)
    at fn (bootstrap:150)
    at Module../src/services/gitlabClient.js (gitlabClient.js:1)
    at __webpack_require__ (bootstrap:784)
    at fn (bootstrap:150)
    at Module../src/App.js (App.css?4433:45)
    at __webpack_require__ (bootstrap:784)
    at fn (bootstrap:150)
    at Module../src/index.js (index.css?f3f6:45)
    at __webpack_require__ (bootstrap:784)
    at fn (bootstrap:150)
    at Object.1 (gitlabClient.js:4)
    at __webpack_require__ (bootstrap:784)
    at checkDeferredModules (bootstrap:45)
    at Array.webpackJsonpCallback [as push] (bootstrap:32)
    at main.chunk.js:1
'''

**Comment by inkidu** [comment]
Updated to v18.1.0 and got a slightly different error. I know you haven't merged your fix. This is informational only.

```
import Gitlab from '@gitbeaker/node';

export default class GitlabClient {

    static instance = null;
    static getInstance() {
        if (this.instance === null) {
            this.instance = new GitlabClient();
        }
        return this.instance;
    }

    constructor() {
        this.config = require('../config.json');
        // this.api = new Gitlab ({
        //     host: this.config.gitlabURL,
        //     token: this.config.gitlabAuthToken,
        // });
        console.log('GitlabClient created!');

        return this;
    }
    
    // async getProjects() {
    //     let projects = await this.api.Projects.all({maxPages: 2});
    //     return projects;
    // }
}
```
Results:
```
Uncaught TypeError: Cannot destructure property 'stat' of '__webpack_require__(...).promises' as it is undefined.
    at Object.<anonymous> (1.chunk.js:13828)
    at Object../node_modules/cacheable-lookup/source/hosts-resolver.js (1.chunk.js:13954)
    at __webpack_require__ (bundle.js:785)
    at fn (bundle.js:151)
    at Object../node_modules/cacheable-lookup/source/index.js (1.chunk.js:13980)
    at __webpack_require__ (bundle.js:785)
    at fn (bundle.js:151)
    at Object../node_modules/got/dist/source/index.js (1.chunk.js:21438)
    at __webpack_require__ (bundle.js:785)
    at fn (bundle.js:151)
    at Module.<anonymous> (1.chunk.js:3221)
    at Module../node_modules/@gitbeaker/node/dist/index.es.js (1.chunk.js:3436)
    at __webpack_require__ (bundle.js:785)
    at fn (bundle.js:151)
    at Module../src/services/gitlabClient.js (main.chunk.js:3834)
    at __webpack_require__ (bundle.js:785)
    at fn (bundle.js:151)
    at Module../src/App.js (main.chunk.js:152)
    at __webpack_require__ (bundle.js:785)
    at fn (bundle.js:151)
    at Module../src/index.js (main.chunk.js:3691)
    at __webpack_require__ (bundle.js:785)
    at fn (bundle.js:151)
    at Object.1 (main.chunk.js:3873)
    at __webpack_require__ (bundle.js:785)
    at checkDeferredModules (bundle.js:46)
    at Array.webpackJsonpCallback [as push] (bundle.js:33)
    at main.chunk.js:1
```

**Comment by jdalrymple** [comment]
What does your tsconfig look like?

**Comment by inkidu** [comment]
I don't have a tsconfig. My project is javascript not typescript.

**Comment by jdalrymple** [comment]
Ok, ill change my local test to see if the error pops up

**Comment by jetersen** [comment]
Based on the test name I presume your clearing the environment variables. Hence you accidentally cleared PATH env.

**Comment by inkidu** [comment]
I have found the enemy and it is me. I switched to @gitbeaker/browser, which I should have been using the entire time, and the workaround you identified:
```
import gitbeaker from '@gitbeaker/node'

const { gitlab } = gitbeaker;
```
is working for me. Thank you for the help.

**Comment by jdalrymple** [comment]
:rocket: Issue was released in [`22.0.0`](https://github.com/jdalrymple/gitbeaker/releases/tag/22.0.0) :rocket:
